src/VirtualClient/VirtualClient.Dependencies.UnitTests/BufferTimeWaiterTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Dependencies
+{
+    using System;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Moq;
+    using NUnit.Framework;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    [TestFixture]
+    [Category("Unit")]
+    internal class BufferTimeWaiterTests
+    {
+        private MockFixture mockFixture;
+
+        public void SetupDefaults(PlatformID platform, Architecture architecture = Architecture.X64)
+        {
+            this.mockFixture = new MockFixture();
+            this.mockFixture.Setup(platform, architecture);
+        }
+
+        [Test]
+        [TestCase(PlatformID.Unix, Architecture.X64)]
+        public async Task BufferTimeWaiterWaitsForExpectedAmountOfTime(PlatformID platform, Architecture architecture)
+        {
+            this.SetupDefaults(platform, architecture);
+            this.mockFixture.Parameters[nameof(BufferTimeWaiter.BufferTimeInSec)] = 1;
```

Comment:
Once you change the wait duration to milliseconds, you can change this test to wait for only 50ms and verify that, makes the test run a lot quicker